src/pages/table/table-editable.index.jsx

```diff
@@ -0,0 +1,191 @@
+  const [loading, setLoading] = useState(false);
+  const [tableItems, setTableItems] = useState(distributions);
+  const [columnDefinitions, saveWidths] = useColumnWidths('React-EditableTable-Widths', EDITABLE_COLUMN_DEFINITIONS);
+  const [preferences, setPreferences] = useLocalStorage(
+    'React-EditableDistributionsTable-Preferences',
+    EDITABLE_PREFERENCES
+  );
+  const [itemsSnap, setItemsSnap] = useState(null);
+
+  const persistChanges = () => {
+    setTableItems(tableItems);
+    setItemsSnap(null);
+  };
+
+  const { items, actions, filteredItemsCount, collectionProps, filterProps, paginationProps, allPageItems } =
+    useCollection(tableItems, {
+      filtering: {
+        empty: <TableEmptyState resourceName="Distribution" />,
+        noMatch: <TableNoMatchState onClearFilter={() => actions.setFiltering('')} />,
+      },
+      pagination: { pageSize: preferences.pageSize },
+      sorting: { defaultState: { sortingColumn: columnDefinitions[0] } },
+      selection: {},
+    });
+
+  const tablePaginationProps = {
+    ...paginationProps,
+    onChange: withSideEffect(paginationProps.onChange, persistChanges),
+  };
+
+  const tableFilterProps = {
+    ...filterProps,
+    onChange: withSideEffect(filterProps.onChange, persistChanges),
+  };
+
+  const tableCollectionProps = {
+    ...collectionProps,
+    onSortingChange: withSideEffect(collectionProps.onSortingChange, persistChanges),
+  };
+
+  const onRefresh = async () => {
+    setLoading(true);
+    await fakeDataFetch(500);
+    persistChanges();
+    setLoading(false);
+  };
+
+  const refreshButtonProps = { onClick: onRefresh };
+
+  const handleSubmit = async (currentItem, column, value) => {
+    await new Promise(resolve => setTimeout(resolve, 1500));
+    let fullCollection = tableItems;
+
+    serverSideErrorsStore.delete(currentItem);
+
+    if (column.id === 'domainName' && !domainNameRegex.test(value)) {
+      serverSideErrorsStore.set(currentItem, INVALID_DOMAIN_MESSAGE);
+      throw new Error('Inline error');
+    }
+
+    const newItem = { ...currentItem, [column.id]: value };
+
+    if (collectionProps.sortingColumn === column) {
+      actions.setSorting(null);
+      fullCollection = allPageItems;
+    }
+
+    if (filterProps.filteringText) {
+      fullCollection = tableItems;
+    }
+
+    if (collectionProps.sortingColumn === column || filterProps.filteringText) {
+      setItemsSnap(items.map(item => (item === currentItem ? newItem : item)));
+    }
+
+    setTableItems(fullCollection.map(item => (item === currentItem ? newItem : item)));
+  };
+
+  return (
+    <Table
+      {...tableCollectionProps}
+      columnDefinitions={columnDefinitions}
+      visibleColumns={preferences.visibleContent}
+      items={itemsSnap || items}
+      submitEdit={handleSubmit}
+      ariaLabels={distributionEditLabels}
+      variant="full-page"
+      stickyHeader={true}
+      resizableColumns={true}
```

Comment:
minor: Which style do we prefer? Because down below we use simply `stickyNotifications` instead of `stickyNotifications={true}`

Reply:
In the main repo, we require the 2nd style with a linter rule: https://github.com/cloudscape-design/components/blob/main/.eslintrc#L31

Reply:
I personally prefer the shorter syntax, but I updated `stickyNotifications` to use the `={true}` style to be consistent with the rest of the code now.